credential_provider/pod_identity_credential_provider.go

```diff
@@ -2,90 +2,29 @@ package credential_provider
 
 import (
 	"context"
-	"encoding/json"
-	"fmt"
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/aws/credentials"
-	"io"
+	"net/http"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials/endpointcreds"
 	authv1 "k8s.io/api/authentication/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	k8sv1 "k8s.io/client-go/kubernetes/typed/core/v1"
-	"k8s.io/klog/v2"
-	"net/http"
-	"strings"
-	"time"
 )
 
 const (
-	podIdentityAudience   = "pods.eks.amazonaws.com"
-	defaultIPv4Endpoint   = "http://169.254.170.23/v1/credentials"
-	defaultIPv6Endpoint   = "http://[fd00:ec2::23]/v1/credentials"
-	httpTimeout           = 100 * time.Millisecond
-	podIdentityAuthHeader = "Authorization"
+	podIdentityAudience = "pods.eks.amazonaws.com"
+	defaultIPv4Endpoint = "http://169.254.170.23/v1/credentials"
+	defaultIPv6Endpoint = "http://[fd00:ec2::23]/v1/credentials"
+	httpTimeout         = 100 * time.Millisecond
 )
 
 var (
 	podIdentityAgentEndpointIPv4 = defaultIPv4Endpoint
 	podIdentityAgentEndpointIPv6 = defaultIPv6Endpoint
 )
 
-// endpointPreference represents the preferred IP address type for Pod Identity Agent endpoint
```

Comment:
This string to iota to decision logic ended up being more complex than necessary. See comments below for more